### ISSUE TITLE
Discussion: Dim Labels

### ISSUE THREAD

**Issue by srush** [harvardnlp/namedtensor#49]
Some folks have suggested we might want to add optional dim labels as well. I was skeptical of this at first, but it might be worthwhile. For instance if you have sequence of word at each index, or the names of the final classes that might really help debugging. 

Syntax proposal: 

```
q = ntorch.ones(5, 7, names=("a", "b"), labels={"a": ["*pad*", "x", "y", "z", "*pad*"]})
```
